### PR TITLE
Add -j opt to filter output by JID

### PR DESCRIPTION
**Pull request by zeylos**
Hello,

As said in the title, I added the -j option to filter output by JID number.
I'm not a developer so my PR may be wrong (or even garbage), any criticism is welcomed.

Anyway thanks for the good work !

### CODE REVIEW

File: checkrestart.c [hunk 1]
```diff
@@ -187,7 +193,7 @@ main(int argc, char *argv[])
 		return (EXIT_FAILURE);
 	}
 
-	while ((ch = getopt(argc, argv, "bHw")) != -1) {
+	while ((ch = getopt(argc, argv, "bHwj:")) != -1) {
```
Comment (Freaky) [inline checkrestart.c:196]: Should be sorted alphabetically, so `"bHj:w"`

File: checkrestart.c [hunk 2]
```diff
@@ -198,6 +204,9 @@ main(int argc, char *argv[])
 		case 'w':
 			termwidth = 0;
 			break;
+		case 'j':
+			jid = atoi(optarg);
```
Comment (Freaky) [inline checkrestart.c:208]: From atoi(3):

> The atoi() function has been deprecated by strtol() and should not be used in new code.

There's a `parse_int()` function in checkrestart to make it easier to handle correctly, e.g:

```c
if (!parse_int(optarg, &jid) || jid < 0) {
    usage();
}
```

File: checkrestart.c (hunk 1 above)
Comment (Freaky) [reply checkrestart.c:196]: (And sort the switch the same way)

**Review by Freaky — COMMENTED** [review]
A couple of nits, nothing serious.  Not like Github breaking repeatedly trying to submit this review :P

**Comment by Freaky** [comment]
Will make tweaks myself.  Thanks!

**Comment by zeylos** [comment]
Ah my bad I didn't know atoi was deprecated.
Thanks a lot for the explainations !

**Pull request merged**